### PR TITLE
ninja: make python a build-only dependency

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

See discussion at #86250.

### CODE REVIEW

**Comment by carlocab** [comment]
Tested locally by building `c-ares` with `ninja` after uninstalling `python@3.9`.

**Comment by carlocab** [comment]
[This](https://github.com/Homebrew/homebrew-core/pull/86257/checks?check_run_id=3769505439#step:7:46) is weird:
```
==> Running Formulae#formula!(c-ares)
==> Determining dependencies...
==> brew fetch --retry cmake gdbm mpdecimal openssl@1.1 pkg-config python@3.9 readline sphinx-doc sqlite xz
```
I think this is related to the Slack discussion about build dependencies of build dependencies being installed unnecessarily. CC @Bo98

**Comment by Bo98** [comment]
Looks like a regression in `brew`. I'll have a look.

**Comment by Bo98** [comment]
Ah wait I think that's intentional behaviour. I'll look into tweaking it however.

**Comment by Bo98** [comment]
Basically `brew deps --include-build` is recursive while `--include-test` is not. This only affects the fetch stage - it shouldn't be actually installed.

File: Formula/ninja.rb [hunk 1]
```diff
@@ -31,6 +30,9 @@ def install
     bin.install "ninja"
     bash_completion.install "misc/bash-completion" => "ninja-completion.sh"
     zsh_completion.install "misc/zsh-completion" => "_ninja"
+    doc.install "doc/manual.asciidoc"
+    elisp.install "misc/ninja-mode.el"
+    (share/"vim/vimfiles/syntax").install "misc/ninja.vim"
```
Comment (carlocab) [inline Formula/ninja.rb:35]: These things seemed useful to install, but probably not worth a `revision` bump.

File: Formula/ninja.rb [hunk 2]
```diff
@@ -20,9 +20,8 @@ class Ninja < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "5ec37aa6d5145de2f396926c75891b578883e2a078989969c011548cae2005cf"
   end
 
-  depends_on "python@3.9"
-
-  uses_from_macos "curl" => :test
```
Comment (carlocab) [inline Formula/ninja.rb:25]: @iMichka where did you see the error that made you add this again? The test seems to pass on Linux even without this now.

Comment (iMichka) [reply Formula/ninja.rb:25]: See https://github.com/Homebrew/homebrew-core/pull/84724

Comment (carlocab) [reply Formula/ninja.rb:25]: I did see #84724, which is why I tagged you here. That PR answers what the error was, but it doesn't say anything about where that error came from.

Comment (iMichka) [reply Formula/ninja.rb:25]: I don’t know. I saw an error, I fixed it. I thought it was the right thing to do. I did not investigate the root cause. It just looked like a dependency was missing, so I added it, and the error was gone. Maybe the fix was the wrong one, and if we have a better solution, please go ahead 😉

Comment (Bo98) [reply Formula/ninja.rb:25]: I imagine the issue might happen if curl is installed in the CI but a dependency of curl is uninstalled (homebrew-test-bot is not designed really to rely on formulae like that - hence our issues with using `HOMEBREW_FORCE_BREWED_CURL`).

**Review by Bo98 — APPROVED** [review]

**Pull request closed without merging**